src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/PermissionsManager.java

```diff
@@ -166,20 +167,128 @@ public List<String> getPermissions(Clan clan) {
         return permissions.get(clan.getTag());
     }
 
+
     /**
      * Charge a player some money
      *
+     * @deprecated use {@link PermissionsManager#chargePlayer(OfflinePlayer, double)} instead
      */
+    @Deprecated(since = "2.19.0", forRemoval = true)
     public boolean playerChargeMoney(OfflinePlayer player, double money) {
-        return economy.withdrawPlayer(player, money).transactionSuccess();
+        return chargePlayer(player, money);
+    }
+
+    /**
+     * Charges the specified amount of money from the player's account.
+     * <p>
+     * As the {@link EconomyTransactionEvent.Cause} is not passed, this method won't fire the {@link EconomyTransactionEvent}.
+     * Use this method when you don't need to track the cause or handle custom transaction events.
+     * </p>
+     *
+     * @param player The player whose account will be charged.
+     * @param money  The amount of money to charge.
+     * @see EconomyTransactionEvent
+     */
+    public boolean chargePlayer(OfflinePlayer player, double money) {
+        return chargePlayer(player, money, null);
+    }
+
+    /**
+     * Charges the specified amount of money from the player's account.
+     *
+     * @param player The player whose account will be charged.
+     * @param money  The amount of money to charge.
+     * @param cause  The cause of the transaction.
+     * @return {@code true} if the charge was successful, {@code false} otherwise.
+     * @see EconomyTransactionEvent
+     */
+    public boolean chargePlayer(@NotNull OfflinePlayer player, double money, @Nullable Cause cause) {
+        EconomyResponse response = Objects.requireNonNull(economy, "Can't find economy provider").withdrawPlayer(player, money);
+        if (!response.transactionSuccess()) {
+            return false;
+        } else if (cause == null) {
+            return true;
+        }
+
+        EconomyTransactionEvent event = new EconomyTransactionEvent(player, money, cause, EconomyTransactionEvent.TransactionType.WITHDRAW);
+        getPluginManager().callEvent(event);
+
+        if (event.isCancelled()) {
+            economy.depositPlayer(player, event.getAmount());
```

Comment:
We withdraw `money` (line 206), but deposit `event.getAmount()`. What if the value changed in the event?

Reply:
That'a a developers choice. I mean, the code does exactly what you write, it could be not expecting to you, but that's a programming. 

if the event was cancelled after assigning a new value to the event, it will deposit even more money. 

Do you think it's confusing?

Reply:
It's an inconsistent logic. Look at lines 221-228. `setAmount` differences are handled there, but not on line 217.

Reply:
What do you mean? It handles. 
As I said in my example: if you caught EconomyTransactionEvent, then `event.setAmount()`, then `event.setCancelled(true)`, it _would_ handle changed amount.

Reply:
Ah, I got it.
If the vaule is smaller than it was, then it won't withdraw.

Reply:
So, what do you suggest?
Extracting handling or just pass `money` instead of `event.getAmount()` as it was suggested by the PR's author

Reply:
> just pass `money`

:+1: